src/basic/path-util.c

```diff
@@ -853,3 +854,28 @@ char *prefix_root(const char *root, const char *path) {
         strcpy(p, path);
         return n;
 }
+
+bool path_prefix_in_list(const char *path, const char *x, ...) {
+        const char *p;
+        va_list ap;
+
+        if (!path || !x)
+                return false;
+
+        va_start(ap, x);
+
+        if (path_startswith(path, x)) {
+                va_end(ap);
+                return true;
+        }
+
+        while ((p = va_arg(ap, const char*)))
+                if (path_startswith(path, p)) {
+                        va_end(ap);
+                        return true;
+                }
+
+        va_end(ap);
+
+        return false;
+}
```

Comment:
Sure, that should work just fine. APIs should have all mandatory parameters listed as variables, but all optional ones as `...`. The only exception is with functions that only have optional ones. In this case, a dummy variable has to be added, such as in `strjoin()`.

If you call `va_start(ap, path);`, and the first optional parameter after `path` is `NULL`, `va_arg()` will return `NULL` in its first invocation.